Wikipedia/Code/WMFThemeableNavigationController.m

```diff
@@ -34,30 +34,19 @@ - (UIStatusBarStyle)preferredStatusBarStyle {
 
 - (void)applyTheme:(WMFTheme *)theme {
     self.theme = theme;
-    self.navigationBar.barTintColor = theme.colors.chromeBackground;
     self.navigationBar.translucent = NO;
     self.navigationBar.tintColor = theme.colors.chromeText;
-    UIImage *backgroundImage = nil;
-    switch (self.style) {
-        case WMFThemeableNavigationControllerStyleEditor:
-            backgroundImage = theme.editorNavigationBarBackgroundImage;
-            break;
-        case WMFThemeableNavigationControllerStyleSheet:
-            backgroundImage = theme.sheetNavigationBarBackgroundImage;
-            break;
-        case WMFThemeableNavigationControllerStyleGallery:
-            self.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
-            [self setNavigationBarHidden:YES animated:NO];
-            break;
-        default:
-            backgroundImage = theme.navigationBarBackgroundImage;
-            break;
+    UINavigationBarAppearance *appearance = [UINavigationBarAppearance appearanceForTheme:theme style:self.style];
```

Comment:
Note this is going off of advice found here: https://developer.apple.com/forums/thread/682420